### DISCUSSION TITLE
Clarification on format => ['json'] to match any, or without a .json suffix

### DISCUSSION THREAD

**Discussion by namotco** [mojolicious/mojo#1792] [Q&A]
My APIs are currently called like so:

"/api/v5/some_endpoint" or also "/api/v5/some_endpoint.json".  I use `$c->respond_to( { any => { json => $data } })`.
I'm trying to alter my routes so they work with the changes introduced in 9.11 (https://github.com/mojolicious/mojo/blob/main/Changes):
```
  $r->get('/foo')->to('bar#yada');
    becomes
      $r->get('/foo' => [format => ['html', 'json']])->to('bar#yada', format => undef);
```
When I try this, 
`$r->get('/api/v5/some_endpoint' => [format => ['json']])->to('api#endpoint', format => undef);`

"/api/v5/some_endpoint" returns a 404.  Is there a way to preserve that besides[format => 1]?

**Comment by Grinnz** [comment]
The optional format mechanism (`format => undef`) only works since 9.11. If you want to support optional formats before then, you can either turn format detection on for the route so it works as before, or create two routes with the same action:

```
$r->get('/api/v5/some_endpoint' => [format => ['json']])->to('api#endpoint');
$r->get('/api/v5/some_endpoint')->to('api#endpoint');
```
